### PR TITLE
Create E2E test suite for accounts and balances

### LINKED ISSUE #296: Implement E2E Test Suite for Balances Pallet
Implement E2E tests covering key Balances pallet flows:

- **Transfer Operations**: Test `transfer_allow_death()`, `transfer_keep_alive()`, `force_transfer()`, and `transfer_all()` with various account states and balance conditions
- Test `force_set_balance()` and `force_unreserve()` for administrative balance operations
- **Account Lifecycle**: Test account creation/reaping via `transfer_allow_death()` when balances cross the existential deposit threshold
- **System Operations**: Test `burn()`, `upgrade_accounts()`, and `force_adjust_total_issuance()` for system-level balance management
- **Edge Cases**: Test zero transfers, self-transfers, maximum balance scenarios, and integration with staking locks (via Currency trait) and governance voting

### PR DESCRIPTION
**Pull request by rockbmb**
Closes #296 .

### CODE REVIEW

**Comment by xlc** [comment]
not actually using it?

**Comment by rockbmb** [comment]
@xlc I know - forgot to submit as draft.

**Comment by rockbmb** [comment]
@xlc I apologize for the size of the PR; I still plan to add a bit more coverage, but nothing that stops it from being reviewed.

File: packages/shared/src/accounts.ts [hunk 1]
```diff
@@ -0,0 +1,3890 @@
...
+
+  const errorEvent = events.find((record) => {
+    const { event } = record
+    return event.section === 'system' && event.method === 'ExtrinsicFailed'
+  })
+  assert(client.api.events.system.ExtrinsicFailed.is(errorEvent!.event))
+  const errorEventData = errorEvent!.event.data
+  assert(errorEventData.dispatchError.isToken)
+  const tokenError = errorEventData.dispatchError.asToken
+  expect(tokenError.isFrozen).toBeTruthy()
```
Comment (xlc) [inline packages/shared/src/accounts.ts:927]: a checkEvent toMatchSnapshots should be enough

File: packages/shared/src/accounts.ts [hunk 2]
```diff
@@ -0,0 +1,3890 @@
...
+    const { event } = record
+    return event.section === 'system' && event.method === 'NewAccount'
+  })
+  expect(newAccountEvent).toBeUndefined()
+
+  const dustLostEvent = events.find((record) => {
+    const { event } = record
+    return event.section === 'balances' && event.method === 'DustLost'
+  })
+  expect(dustLostEvent).toBeUndefined()
```
Comment (xlc) [inline packages/shared/src/accounts.ts:990]: this can be done by checkEvents toMatchSnapshot and expect the snapshot to be empty array

**Comment by rockbmb** [comment]
@xlc in a follow-up PR, I'm planning to enable these tests for more chains: Acala, Astar, Moonbeam and Bifrost, at least.

The CI pipeline is growing slowly but surely; let me know when it's time to have a conversation about infrastructure costs and long-term sustainability regarding all these contributions.

**Comment by xlc** [comment]
Right now everything are run on free public github server so there isn't much infra cost. the only cost is AI reviewer, which is just an experiential thing and cost ~ $0.10 - $0.50 per review
If we are going to say double the number of the tests, the free one should still be enough but might approaching some limit and we could consider some alternatives
